kairos-l1-utils/src/lib.rs

```diff
@@ -0,0 +1,109 @@
+use casper_client::{
+    get_state_root_hash, query_global_state, types::StoredValue, JsonRpcId,
+    Verbosity,
+};
+use casper_hashing::Digest;
+use casper_types::URef;
+
+use casper_client::{
+    rpcs::results::PutDeployResult,
+    types::{Deploy, DeployBuilder, ExecutableDeployItem, Timestamp},
+    SuccessResponse,
+};
+use casper_types::{crypto::SecretKey, RuntimeArgs};
+use std::fs;
+
+pub const DEFAULT_PAYMENT_AMOUNT: u64 = 1_000_000_000_000;
+
+pub async fn install_wasm_bytecode(
+    node_address: &str,
+    rpc_port: &str,
+    chain_name: &str,
+    runtime_args: RuntimeArgs,
+    module_bytes: &[u8],
+    secret_key_path: &str,
+) -> SuccessResponse<PutDeployResult> {
+    let session: ExecutableDeployItem =
+        ExecutableDeployItem::new_module_bytes(module_bytes.into(), runtime_args);
+    let secret_key_bytes: Vec<u8> = fs::read(secret_key_path).unwrap();
+    let secret_key: SecretKey = SecretKey::from_pem(secret_key_bytes.clone()).unwrap();
+
+    let deploy: Deploy = DeployBuilder::new(chain_name, session, &secret_key)
+        .with_timestamp(Timestamp::now())
+        .with_standard_payment(DEFAULT_PAYMENT_AMOUNT)
+        .build()
+        .unwrap();
+
+    casper_client::put_deploy(
+        JsonRpcId::String(rpc_port.to_string()),
+        node_address,
+        casper_client::Verbosity::Low,
+        deploy,
+    )
+    .await
+    .unwrap()
+}
+
+pub async fn query_state_root_hash(node_address: &str, rpc_port: &str) -> Digest {
+    get_state_root_hash(
+        JsonRpcId::String(rpc_port.to_owned()),
```

Comment:
I'm pretty sure this is not the way the port should be used?

Reply:
It's pretty weird and doesn't seem to do anything... The way I am doing it works but I am also confused so I'll try to figure out how they intended it.

Reply:
"It works" is not reasonable. The rpc id is used for matching requests and responses as they are not guaranteed to arrive in a specific order. Please just use an arbitrary id, instead of mispurposing the port which is semantically wrong.

Reply:
I accidentally clicked on resolve earlier.

If it is an arbitrary ID that we don't need / use then what I'm currently doing will never cause a problem, we can replace it with "0" but does it matter?

Reply:
It does, because in your function you expect an `rpc_port` as an argument that you don't even use later on when you query the state root hash. You instead use it as an `rpc_id`. Anyone using or maintaining your code will be confused about the purpose this argument serves

Reply:
I'll remove and default the id to 0